test/src/test/java/hudson/security/HudsonPrivateSecurityRealmFIPSTest.java

```diff
@@ -149,6 +148,87 @@ private static void userCreationWithJBCryptPasswordsStep(JenkinsRule j) throws E
                 is("The hashed password was hashed with an incorrect algorithm. Jenkins is expecting $PBKDF2"));
     }
 
+    @Test
+    public void validatePasswordLengthForFIPS() throws Throwable {
+        rjr.then(HudsonPrivateSecurityRealmFIPSTest::validatePasswordLengthForFIPSStep);
+    }
+
+    private static void validatePasswordLengthForFIPSStep(JenkinsRule j) throws Exception {
+        HudsonPrivateSecurityRealm securityRealm = new HudsonPrivateSecurityRealm(false, false, null);
+        j.jenkins.setSecurityRealm(securityRealm);
+
+        User u1 =  securityRealm.createAccount("test", "password");
```

Comment:
how is this possible? this password is invalid in FIPS mode (shorter than 14 characteres).

Reply:
https://cloudbees.atlassian.net/browse/BEE-53803

Created a ticket for this as discussed.
We also need to discuss about potential impact this ticket can have.

Reply:
This is because FIPS compliant encryption for PKKDF is not in use.  This FIPS test merley sets the flag.  if you use a FIPS compiant implementation of PBKDF2 you should get an exception here.

@olamy added some code to the jenkins-test-harness to use FIPS compiant providers in tests for FIPS so if you swap to that it should blow up.  
it is still likely there is missing validation for the direct admin to create a user, but that should just be missing validation.

In any case - you should assume that you can not create this user with a short password here, create the user with an ok password and then try to change it to an invalid password.